src/output.rs

```diff
@@ -1,175 +1,290 @@
+    #[cfg(unix)]
+    fn print_entry_uncolorized(&mut self, entry: &DirEntry) -> io::Result<()> {
+        use std::os::unix::ffi::OsStrExt;
+
+        if self.config.interactive_terminal || self.config.path_separator.is_some() {
+            // Fall back to the base implementation
+            self.print_entry_uncolorized_base(entry)
+        } else {
+            // Print path as raw bytes, allowing invalid UTF-8 filenames to be passed to other processes
+            self.stdout
+                .write_all(entry.stripped_path(self.config).as_os_str().as_bytes())?;
+            self.print_trailing_slash(entry, None)
+        }
     }
-    write!(stdout, "{path_string}")?;
-    print_trailing_slash(stdout, entry, config, None)
-}
 
-#[cfg(not(unix))]
-fn print_entry_uncolorized<W: Write>(
-    stdout: &mut W,
-    entry: &DirEntry,
-    config: &Config,
-) -> io::Result<()> {
-    print_entry_uncolorized_base(stdout, entry, config)
-}
+    fn print_entry_json_obj(&mut self, detail: &FileDetail) -> io::Result<()> {
+        write!(self.stdout, "{{")?;
 
-#[cfg(unix)]
-fn print_entry_uncolorized<W: Write>(
-    stdout: &mut W,
-    entry: &DirEntry,
-    config: &Config,
-) -> io::Result<()> {
-    use std::os::unix::ffi::OsStrExt;
-
-    if config.interactive_terminal || config.path_separator.is_some() {
-        // Fall back to the base implementation
-        print_entry_uncolorized_base(stdout, entry, config)
-    } else {
-        // Print path as raw bytes, allowing invalid UTF-8 filenames to be passed to other processes
-        stdout.write_all(entry.stripped_path(config).as_os_str().as_bytes())?;
-        print_trailing_slash(stdout, entry, config, None)
+        match &detail.path {
+            PathEncoding::Utf8(path_utf8) => {
+                write!(self.stdout, "\"path\":\"{}\"", path_utf8)?;
+            }
+            PathEncoding::Bytes(path_bytes) => {
+                write!(
+                    self.stdout,
+                    "\"path_b64\":\"{}\"",
+                    BASE64_STANDARD.encode(path_bytes)
+                )?;
+            }
+        }
+
+        write!(self.stdout, ",\"type\":\"{}\"", detail.file_type)?;
+
+        if let Some(size) = detail.size {
+            write!(self.stdout, ",\"size\":{size}")?;
+        }
+        if let Some(mode) = detail.mode {
+            write!(self.stdout, ",\"mode\":{mode:o}")?;
+        }
+        if let Some(modified) = &detail.modified {
+            write!(self.stdout, ",\"modified\":\"{}\"", modified)?;
+        }
+        if let Some(accessed) = &detail.accessed {
+            write!(self.stdout, ",\"accessed\":\"{}\"", accessed)?;
+        }
+        if let Some(created) = &detail.created {
+            write!(self.stdout, ",\"created\":\"{}\"", created)?;
+        }
+        write!(self.stdout, "}}")
+    }
+
+    fn print_entry_detail(&mut self, format: OutputFormat, entry: &DirEntry) -> io::Result<()> {
```

Comment:
nit: We could potentially combine this with the  print_entry_json_obj method now, since it is only used for jsonl. 

OTOH, it may be useful in the future if we add a yaml format later, or maybe a tabular format?